cpp/include/raft/util/cuda_utils.cuh

```diff
@@ -516,6 +516,16 @@ struct Nop {
   HDI Type operator()(Type in, IdxType i = 0) { return in; }
 };
 
+template <typename Type, typename IdxType = int>
+struct SqrtOp {
```

Comment:
Here, and in other "Functor" structs: would you consider moving the template parameters onto the `operator()` where possible? This way, the template parameter will be inferred automatically - less typing and opportunities to introduce bugs on the user side.